android/app/src/main/java/org/fossasia/openevent/activities/SearchActivity.java

```diff
@@ -57,25 +62,34 @@ protected void onCreate(Bundle savedInstanceState) {
 
         getSupportActionBar().setDisplayHomeAsUpEnabled(true);
 
-        handleVisibility();
-
-        globalSearchAdapter = new GlobalSearchAdapter(results, this);
-        searchRecyclerView.setAdapter(globalSearchAdapter);
-        searchRecyclerView.addItemDecoration(new DividerItemDecoration(this, DividerItemDecoration.VERTICAL));
-        searchRecyclerView.setLayoutManager(new LinearLayoutManager(this));
+        setUpRecyclerView();
 
         if (savedInstanceState != null && savedInstanceState.getString(SEARCH) != null) {
             searchText = savedInstanceState.getString(SEARCH);
         }
-
         if (Intent.ACTION_SEARCH.equals(getIntent().getAction())) {
+
             handleIntent(getIntent());
         }
+        searchViewModel = ViewModelProviders.of(this).get(SearchActivityViewModel.class);
+        searchText = searchViewModel.getSearchText();
+        if(searchViewModel.getSearchResults()!=null) {
+            searchViewModel.getSearchResults().observe(SearchActivity.this, searches -> {
+                results.clear();
+                results.addAll(searches);
+                globalSearchAdapter.notifyDataSetChanged();
+                if (!Utils.isEmpty(searchText)) {
+                    searchQuery(searchText);
+                }
+             });
+        }
+        handleVisibility();
+
     }
 
     @Override
     protected int getLayoutResource() {
-        return R.layout.activity_search;
+       return R.layout.activity_search;
```

Comment:
Indentation is off, infact don't touch the part of the code which your issue isn't associated with.